hooks/useMqlQuery.ts

```diff
@@ -263,23 +265,25 @@ export default function useMqlQuery({
     variables: {
       queryId: state.queryId || "",
     },
-    pause: !state.queryId || state.cancelledQueries.includes(state.queryId),
+    pause:
+      skip || !state.queryId || state.cancelledQueries.includes(state.queryId),
   });
 
-  // I am creating this!
   useEffect(() => {
     if (error) {
       dispatch({
         type: "fetchResultsFail",
         errorMessage: getErrorMessage(error),
       });
       handleCombinedError(error);
+      return;
```

Comment:
@jjrajani this will prevent the flickering. Before, it was encountering an error while the query was still running. So it would render the error state and then dispatch a fetch shortly after, returning it to a loading state, and so on until the query completes.